zabdig

```diff
@@ -302,7 +302,9 @@ class ZHost(ZBaseHost):
     def __init__(self, hostinfo, proxies):
         super(ZHost, self).__init__(hostinfo)
 
-        if proxies and hostinfo['proxy_hostid'] != '0':
+        if proxies and hostinfo['proxyid'] != '0':
+            self.proxy = proxies[hostinfo['proxyid']]
+        elif proxies and hostinfo['proxy_hostid'] != '0':
```

Comment:
I suggest keeping just proxyid here. And dropping the proxy_hostid options.

See suggested changes below.